android/src/test/java/org/mozilla/mozstumbler/client/UpdaterTest.java

```diff
@@ -90,9 +92,11 @@ public void testUpdaterThrottlesRequests() {
         ServiceLocator.getInstance().putService(IHttpUtil.class, mockHttp);
         ServiceLocator.getInstance().putService(ISystemClock.class, clock);
 
-        // wifi is always unavailable
-        doReturn(false).when(upd).wifiExclusiveAndUnavailable(Mockito.any(Context.class));
+        ConnectivityManager connectivityManager = (ConnectivityManager) ctx.getSystemService(Context.CONNECTIVITY_SERVICE);
+        MyShadowConnectivityManager shadowConnManager = (MyShadowConnectivityManager) Robolectric.shadowOf(connectivityManager);
 
+        shadowConnManager.setConnectedFlag(true);
+        doReturn(false).when(upd).wifiExclusiveAndUnavailable(Mockito.any(Context.class));
```

Comment:
It's not clear to me what these changes are adding.  Can we add some assertions that demonstrate why the extra shadow is required?